UefiCpuPkg/SecCore/SecMain.c

```diff
@@ -282,31 +164,9 @@ SecStartup (
   //
   InitializeFloatingPointUnits ();
 
-  // |-------------------|---->
-  // |IDT Table          |
-  // |-------------------|
-  // |PeiService Pointer |    PeiStackSize
-  // |-------------------|
-  // |                   |
-  // |      Stack        |
-  // |-------------------|---->
-  // |                   |
-  // |                   |
-  // |      Heap         |    PeiTemporayRamSize
-  // |                   |
-  // |                   |
-  // |-------------------|---->  TempRamBase
-
-  IdtTableInStack.PeiService = 0;
-  for (Index = 0; Index < SEC_IDT_ENTRY_COUNT; Index++) {
-    ZeroMem ((VOID *)&IdtTableInStack.IdtTable[Index], sizeof (IA32_IDT_GATE_DESCRIPTOR));
-    CopyMem ((VOID *)&IdtTableInStack.IdtTable[Index], (VOID *)&mIdtEntryTemplate, sizeof (UINT64));
-  }
-
-  IdtDescriptor.Base  = (UINTN)&IdtTableInStack.IdtTable;
-  IdtDescriptor.Limit = (UINT16)(sizeof (IdtTableInStack.IdtTable) - 1);
-
-  AsmWriteIdtr (&IdtDescriptor);
+ #if defined (MDE_CPU_IA32) || defined (MDE_CPU_X64)
+  IdtTableSetup ();
+ #endif

```

Comment:
You have to move IdtTableSetup() implementation in SecStartup().
Your new change cannot work for IA32/X64.

The storage of IDT table can use stack because the code never returns from SecStartup().